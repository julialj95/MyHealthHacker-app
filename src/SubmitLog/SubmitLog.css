.log_form_header {
  font-size: 22px;
}
.new_log {
  text-align: center;
  background-color: #6e3667;
  margin-left: 10%;
  margin-right: 10%;
  border: solid 5px #88d317;
  border-radius: 15px;
  color: white;
}
.log_section {
  display: flex;
  flex-direction: row;
}

@media all and (min-width: 600px) {
  .log_form_header {
    font-size: 24px;
  }
}

@media all and (min-width: 768px) {
}

@media all and (min-width: 960px) {
  .log_form_header {
    font-size: 27px;
  }
}
