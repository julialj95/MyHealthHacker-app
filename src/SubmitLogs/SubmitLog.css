.new_log {
  text-align: center;
}
.log_section {
  display: flex;
  flex-direction: row;
}
.mental,
.sleep,
.exercise,
.other {
  border: solid 1px black;
  border-radius: 5px;
  flex: 1;
}
