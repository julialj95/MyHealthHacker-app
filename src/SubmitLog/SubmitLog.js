import React, { useState } from "react";
import { useHistory } from "react-router-dom";
import "./SubmitLog.css";
import config from "../config";
import TokenService from "../services/token-service";
import LogInputForm from "../LogInputForm";

function Logs() {
  const [logData, setLogData] = useState({
    log_date: "",
    mood: "",
    stress: "",
    sleep_hours: "",
    sleep_quality: "",
    exercise_minutes: "",
    exercise_type: "",
    water: "",
    notes: "",
  });

  const [error, setError] = useState("");
  const history = useHistory();

  const handleChange = (e) => {
    const { name, value } = e.target;
    setLogData({
      ...logData,
      [name]: value,
    });
  };

  const handleSubmitLog = (e) => {
    e.preventDefault();
    const {
      log_date,
      mood,
      stress,
      sleep_hours,
      sleep_quality,
      exercise_minutes,
      exercise_type,
      water,
      notes,
    } = logData;

    fetch(`${config.API_BASE_URL}/logs`, {
      method: "POST",
      body: JSON.stringify({
        log_date,
        mood,
        stress,
        sleep_hours,
        sleep_quality,
        exercise_minutes,
        exercise_type,
        water,
        notes,
      }),
      headers: {
        "content-type": "application/json",
        Authorization: "Bearer " + TokenService.getAuthToken(),
      },
    })
      .then((res) => {
        if (!res.ok) {
          res.json().then((e) => Promise.reject(e));
        }

        res.json();
        history.push("/logs");
      })
      .catch((error) => setError(error));
  };
  console.log("logData.log_date", logData.log_date);
  return (
    <main className="new_log">
      <h2>Submit Log:</h2>
      <LogInputForm
        handleChange={handleChange}
        handleSubmitLog={handleSubmitLog}
        log_date={logData.log_date}
        mood={logData.mood}
        stress={logData.stress}
        sleep_hours={logData.sleep_hours}
        sleep_quality={logData.sleep_quality}
        exercise_minutes={logData.exercise_minutes}
        exercise_type={logData.exercise_type}
        water={logData.water}
        notes={logData.notes}
      />
      {error ? <h2>{error.message}</h2> : null}
    </main>
  );
}

export default Logs;
