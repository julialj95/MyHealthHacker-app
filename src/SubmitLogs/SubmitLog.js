import React from "react";
import "./SubmitLog.css";

class Logs extends React.Component {
  render() {
    return (
      <main className="new_log">
        <h2>Submit Log:</h2>
        <form>
          <section>
            <label for="date">Current Date</label>
            <input type="date" id="date" name="date" />
          </section>
          <br />

          <div className="log_section">
            <section className="mental">
              <h2>Psychological</h2>
              <div className="mood_rating">
                <h3>Mood Rating</h3>
                <h3>Rate your mood out of 5</h3>
                <input type="radio" id="1" name="mood" value="1" />
                <label for="mood">1</label>
                <input type="radio" id="2" name="mood" value="2" />
                <label for="mood">2</label>
                <input type="radio" id="3" name="mood" value="3" />
                <label for="mood">3</label>
                <input type="radio" id="4" name="mood" value="4" />
                <label for="mood">4</label>
                <input type="radio" id="5" name="mood" value="5" />
                <label for="mood">5</label>
              </div>
              <div className="stress">
                <h2>Stress Level</h2>
                <h3>Rate your stress level with 5 being the most stressed</h3>
                <input type="radio" id="1" name="stress" value="1" />
                <label for="stress">1</label>
                <input type="radio" id="2" name="stress" value="2" />
                <label for="stress">2</label>
                <input type="radio" id="3" name="stress" value="3" />
                <label for="stress">3</label>
                <input type="radio" id="4" name="stress" value="4" />
                <label for="stress">4</label>
                <input type="radio" id="5" name="stress" value="5" />
                <label for="stress">5</label>
              </div>
            </section>
            <section className="sleep">
              <h2>Sleep</h2>
              <label for="sleep">Hours slept</label>
              <br />
              <input type="text" name="sleep_hours" id="sleep_hours" />
              <div>
                <h3>Sleep quality</h3>
                <input type="radio" id="1" name="sleep_quality" value="1" />
                <label for="sleep_quality">1</label>
                <input type="radio" id="2" name="sleep_quality" value="2" />
                <label for="sleep_quality">2</label>
                <input type="radio" id="3" name="sleep_quality" value="3" />
                <label for="sleep_quality">3</label>
                <input type="radio" id="4" name="sleep_quality" value="4" />
                <label for="sleep_quality">4</label>
                <input type="radio" id="5" name="sleep_quality" value="5" />
                <label for="sleep_quality">5</label>
              </div>
            </section>
            <br />
            <section className="exercise">
              <h2>Physical</h2>
              <label for="exercise">Minutes of exercise</label>
              <br />
              <input type="text" name="exercise" id="exercise" />
              <br />
              <br />
              <label for="exercise_type">Type of exercise</label>
              <br />
              <input type="text" name="exercise_type" id="exercise_type" />
              <br />
              <br />

              <label for="water">Ounces of water consumed</label>
              <br />
              <input type="text" name="water" id="water" />
            </section>
          </div>
          <br />
          <br />
          <div className="log_section">
            <section className="other">
              <label for="notes">Additional notes</label>
              <br />
              <textarea id="notes" name="notes" rows="4" cols="50">
                Anything else you would like to track? Note it here!
              </textarea>
            </section>
          </div>
          <section class="button-section">
            <button type="submit">Submit</button>
            <button type="reset">Reset</button>
          </section>
        </form>
      </main>
    );
  }
}
export default Logs;
