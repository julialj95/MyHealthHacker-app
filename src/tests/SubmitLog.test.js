import React from "react";
import ReactDOM from "react-dom";
import SubmitLog from "../SubmitLog/SubmitLog";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
  const div = document.createElement("div");

  ReactDOM.render(
    <Router>
      <SubmitLog />
    </Router>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as expected", () => {
  const tree = renderer
    .create(
      <Router>
        <SubmitLog />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
