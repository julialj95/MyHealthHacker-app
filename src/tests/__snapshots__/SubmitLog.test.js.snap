// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<main
  className="new_log"
>
  <h2>
    Submit Log:
  </h2>
  <form
    onSubmit={[Function]}
  >
    <section>
      <label
        htmlFor="date"
      >
        Current Date
      </label>
      <input
        id="log_date"
        max="2021-04-27"
        name="log_date"
        onChange={[Function]}
        type="date"
        value=""
      />
    </section>
    <br />
    <div
      className="log_section"
    >
      <section
        className="mental"
      >
        <h2>
          Psychological
        </h2>
        <div
          className="mood_rating"
        >
          <h3>
            Mood Rating
          </h3>
          <h3>
            Rate your mood out of 5
          </h3>
          <input
            id="1"
            name="mood"
            onChange={[Function]}
            type="radio"
            value={1}
          />
          <label
            htmlFor="mood"
          >
            1
          </label>
          <input
            id="2"
            name="mood"
            onChange={[Function]}
            type="radio"
            value={2}
          />
          <label
            htmlFor="mood"
          >
            2
          </label>
          <input
            id="3"
            name="mood"
            onChange={[Function]}
            type="radio"
            value={3}
          />
          <label
            htmlFor="mood"
          >
            3
          </label>
          <input
            id="4"
            name="mood"
            onChange={[Function]}
            type="radio"
            value={4}
          />
          <label
            htmlFor="mood"
          >
            4
          </label>
          <input
            id="5"
            name="mood"
            onChange={[Function]}
            type="radio"
            value={5}
          />
          <label
            htmlFor="mood"
          >
            5
          </label>
        </div>
        <div
          className="stress"
        >
          <h2>
            Stress Level
          </h2>
          <h3>
            Rate your stress level with 5 being the most stressed
          </h3>
          <input
            id="1"
            name="stress"
            onChange={[Function]}
            type="radio"
            value={1}
          />
          <label
            htmlFor="stress"
          >
            1
          </label>
          <input
            id="2"
            name="stress"
            onChange={[Function]}
            type="radio"
            value={2}
          />
          <label
            htmlFor="stress"
          >
            2
          </label>
          <input
            id="3"
            name="stress"
            onChange={[Function]}
            type="radio"
            value={3}
          />
          <label
            htmlFor="stress"
          >
            3
          </label>
          <input
            id="4"
            name="stress"
            onChange={[Function]}
            type="radio"
            value={4}
          />
          <label
            htmlFor="stress"
          >
            4
          </label>
          <input
            id="5"
            name="stress"
            onChange={[Function]}
            type="radio"
            value={5}
          />
          <label
            htmlFor="stress"
          >
            5
          </label>
        </div>
      </section>
      <section
        className="sleep"
      >
        <h2>
          Sleep
        </h2>
        <label
          htmlFor="sleep"
        >
          Hours slept
        </label>
        <br />
        <input
          id="sleep_hours"
          name="sleep_hours"
          onChange={[Function]}
          type="number"
          value=""
        />
        <div>
          <h3>
            Sleep quality
          </h3>
          <input
            id="1"
            name="sleep_quality"
            onChange={[Function]}
            type="radio"
            value={1}
          />
          <label
            htmlFor="sleep_quality"
          >
            1
          </label>
          <input
            id="2"
            name="sleep_quality"
            onChange={[Function]}
            type="radio"
            value={2}
          />
          <label
            htmlFor="sleep_quality"
          >
            2
          </label>
          <input
            id="3"
            name="sleep_quality"
            onChange={[Function]}
            type="radio"
            value={3}
          />
          <label
            htmlFor="sleep_quality"
          >
            3
          </label>
          <input
            id="4"
            name="sleep_quality"
            onChange={[Function]}
            type="radio"
            value={4}
          />
          <label
            htmlFor="sleep_quality"
          >
            4
          </label>
          <input
            id="5"
            name="sleep_quality"
            onChange={[Function]}
            type="radio"
            value={5}
          />
          <label
            htmlFor="sleep_quality"
          >
            5
          </label>
        </div>
      </section>
      <br />
      <section
        className="exercise"
      >
        <h2>
          Physical
        </h2>
        <label
          htmlFor="exercise_minutes"
        >
          Minutes of exercise
        </label>
        <br />
        <input
          id="exercise_minutes"
          name="exercise_minutes"
          onChange={[Function]}
          type="number"
          value=""
        />
        <br />
        <br />
        <label
          htmlFor="exercise_type"
        >
          Type of exercise
        </label>
        <br />
        <input
          id="exercise_type"
          name="exercise_type"
          onChange={[Function]}
          type="text"
          value=""
        />
        <br />
        <br />
        <label
          htmlFor="water"
        >
          Ounces of water consumed
        </label>
        <br />
        <input
          id="water"
          name="water"
          onChange={[Function]}
          type="number"
          value=""
        />
      </section>
    </div>
    <br />
    <br />
    <div
      className="log_section"
    >
      <section
        className="other"
      >
        <label
          htmlFor="notes"
        >
          Additional notes
        </label>
        <br />
        <textarea
          cols="50"
          id="notes"
          name="notes"
          onChange={[Function]}
          placeholder="Anything else you would like to track? Note it here!"
          rows="4"
          value=""
        />
      </section>
    </div>
    <section
      className="button-section"
    >
      <button
        type="submit"
      >
        Submit
      </button>
      <button
        onClick={[Function]}
        type="reset"
      >
        Reset
      </button>
    </section>
  </form>
</main>
`;
